### PR TITLE
BAU: Improve docker build & pipeline test environment

### PR DESCRIPTION
**Pull request by whi-tw**
## What

Selenium provide docker images to run selenium nodes. In this PR, I've used this image as a base to run our tests.

The start script (`/auth-docker-entrypoint.sh`) starts up the selenium node, waits for it to be ready, then runs the test, pointed at localhost.

We now fully build the tests, and copy the built jars into the container - this prevents us rebuilding the tests on every test run in the pipeline.

## How to review

Run the tests in the same way the pipeline will:

```sh
docker build . -t acceptance:latest -f Dockerfile.chromium
export AWS_PROFILE=gds-di-development-admin
source ./scripts/check_aws_creds.sh
docker run -p 4442-4444:4442-4444 --privileged --env-file <(aws configure export-credentials --format env-no-export) -it --rm --entrypoint /bin/bash acceptance:latest /auth-docker-entrypoint.sh build
```

### CODE REVIEW

**Review by mattvot — CHANGES_REQUESTED** [review]
I've added a few the optional suggestions as commented.

I can't see the test plan succeedig atm. Running `docker build . -t acceptance:latest -f Dockerfile.chromium` gives me the following error:

```
44.32 Get:104 http://deb.debian.org/debian sid/main arm64 python3-olefile all 0.47-1 [39.3 kB]
44.33 Get:105 http://deb.debian.org/debian sid/main arm64 python3-pil arm64 10.4.0-1+b1 [498 kB]
44.50 Get:106 http://deb.debian.org/debian sid/main arm64 python3-pygments all 2.18.0+dfsg-1 [836 kB]
44.79 Get:107 http://deb.debian.org/debian sid/main arm64 systemd-cryptsetup arm64 257~rc3-1 [158 kB]
44.98 debconf: delaying package configuration, since apt-utils is not installed
45.00 Fetched 171 MB in 32s (5,308 kB/s)
(Reading database ... 18906 files and directories currently installed.)
45.02 Preparing to unpack .../base-files_13.6_arm64.deb ...
45.03 dpkg-divert: error: 'diversion of /lib32 to /.lib32.usr-is-merged by base-files' clashes with 'diversion of /lib32 to /lib32.usr-is-merged by base-files'
45.03 dpkg: error processing archive /var/cache/apt/archives/base-files_13.6_arm64.deb (--unpack):
45.03  new base-files package pre-installation script subprocess returned error exit status 2
45.09 Errors were encountered while processing:
45.09  /var/cache/apt/archives/base-files_13.6_arm64.deb
45.10 E: Sub-process /usr/bin/dpkg returned an error code (1)
```

For what it's worth, after seeing this the first time I ran `docker system prune --all` and updated Docker Desktop. Re-ran and saw the same issue.

🤔

File: Dockerfile.chromium [hunk 1]
```diff
@@ -0,0 +1,54 @@
+FROM gradle:7-jdk17 as builder
```
[the remark this answers is not in the document]
Comment (whi-tw) [reply Dockerfile.chromium:1]: Oops! Fixed

File: Dockerfile.chromium [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    && rm -rf /var/lib/apt/lists/*
+
+USER ${SEL_USER}
+
+COPY --chown=${SEL_USER}:${SEL_GROUP} --from=builder /root/.gradle /home/${SEL_USER}/.gradle
+
+
+WORKDIR /test
+
+COPY --chown=${SEL_USER}:${SEL_GROUP} build.gradle settings.gradle gradlew ./
```
[the remark this answers is not in the document]
Comment (whi-tw) [reply Dockerfile.chromium:34]: yep,valid point - unsure why I did this differently for this specific line!

**Review by whi-tw — COMMENTED** [review]
WHY IS VSCODE MAKING ME DO A REVIEW TO REPLY TO COMMENTS

**Comment by whi-tw** [comment]
> I can't see the test plan succeedig atm. Running `docker build . -t acceptance:latest -f Dockerfile.chromium` gives me the following error:

So, openjdk is already installed. Triggering an install in docker caused other dependencies to update, which caused what you saw!

As it's already there (selenium depends on it), we no longer install it.

**Review by mattvot — APPROVED** [review]
This is working locally for me, though I have to follow a different test plan as calling `source ./scripts/check_aws_creds.sh` directly closes the terminal.

Instead I just did:

```
export AWS_PROFILE=gds-di-development-admin
./run-in-docker.sh build
```

**Pull request closed without merging**